src/libmeasurement_kit/nettests/multi_ndt.cpp

```diff
@@ -0,0 +1,266 @@
+        std::vector<double> good_speeds(
+            // Note: going beyond vector limits would raise
+            // a std::length_error exception
+            speeds.begin() + 6, speeds.end() - 2
+        );
+        double sum = 0.0;
+        for (auto &x : good_speeds) {
+            sum += x;
+        };
+        if (good_speeds.size() <= 0) {
+            logger->warn("The vector of good speeds is empty");
+            return nullptr;
+        }
+        return sum / good_speeds.size();
+    } catch (const std::exception &) {
+        logger->warn("Cannot compute download speed");
+        // FALLTHROUGH
+    }
+    return nullptr;
+}
+
+static report::Entry compute_stats(report::Entry &root, std::string key,
+                                   Var<Logger> logger) {
+    report::Entry stats;
+    report::Entry test_s2c;
+    try {
+        test_s2c = root[key]["test_s2c"][0] /* We know it's just one entry */;
+    } catch (const std::exception &) {
+        logger->warn("cannot access root[\"%s\"][\"test_s2c\"][0]",
+                     key.c_str());
+        return nullptr;  /* Cannot compute this stat */
+    }
+    stats["ping"] = compute_ping(test_s2c, logger);
+    stats["download"] = compute_download_speed(test_s2c, logger);
+    stats["key"] = key;
+    return stats;
+}
+
+static void compute_simple_stats(report::Entry &entry, Var<Logger> logger) {
+    report::Entry single = compute_stats(entry, "single_stream", logger);
+    report::Entry multi = compute_stats(entry, "multi_stream", logger);
+    report::Entry selected;
+
+    /*
+     * Here we basically pick up the fastest of the two tests.
+     */
+    if (single["ping"] != nullptr and multi["ping"] != nullptr) {
+        if (single["download"] != nullptr and multi["download"] != nullptr) {
+            double singled = single["download"];
+            double multid = multi["download"];
+            if (singled > multid) {
+                selected = single;
+            } else {
+                selected = multi;
+            }
+        } else if (single["download"] != nullptr) {
+            logger->warn("Multi-stream download is null");
+            selected = single;
+        } else if (multi["download"] != nullptr) {
+            logger->warn("Single-stream download is null");
+            selected = multi;
+        } else {
+            logger->warn("Single- and multi-stream download are null");
+        }
+    } else if (single["ping"] != nullptr) {
+        logger->warn("Multi-stream ping is null");
+        selected = single;
+    } else if (multi["ping"] != nullptr) {
+        logger->warn("Single-stream ping is null");
+        selected = multi;
+    } else {
+        logger->warn("Single- and multi-stream ping are null");
+    }
+
+    entry["simple"] = selected;
+}
+
+static void compute_advanced_stats(report::Entry &entry) {
+    report::Entry test_s2c =
+        entry["single_stream"]["test_s2c"][0] /* We know it's just one entry */;
```

Comment:
[Late Review] It could increase robustness adding a comment saying that this function can throw exceptions.